### ISSUE TITLE
Uncoverable decision on 'case' when opening bracket is in a new line or is one line case

### ISSUE THREAD

**Issue by czoltan1026** [gcovr/gcovr#682]
**Describe the bug**
The following code produces an uncoverable decisions at cases 0 and 1

```c++
void f(const int x) {

  switch (x) {
    case 0: // uncoverable decision
    {
      break;
    }

    case 1:{break;} // uncoverable decision

    case 2: { // ok
      break;
    }

    default: // ok
      break;
  }
}

int main() {

  f(0);
  f(1);
  f(2);
  f(3);

}
```

**To Reproduce**
g++ -o example example.cpp -O0 --coverage
./example
gcov example
gcovr --decision --html-details --output=coverage.html --object-directory .

**Expected behavior**
Decisions should be coverable at the cases independent of the brace position

**Desktop (please complete the following information):**
 - OS: Windows/linux
 - GCC 9.40, GCC 11.1.0, CLANG 12.0.0
 - GCOVR 5.2

**Comment by Spacetown** [comment]
While investigating in #679 I found out that the parsing hardly depends on the formating style (see fix #681). I think we have to check if there is a better approach. 🤔

**Comment by bence42** [comment]
@Spacetown  #681 does not fix this issue. Case 0 and case 1 is now analyzed but marked as not taken, which is wrong.

**Comment by Spacetown** [comment]
What about the branches in this case?

**Comment by bence42** [comment]
What do you mean?

**Comment by Spacetown** [comment]
The branch coverage. On the level of the switch there should be 4 branches. Is every branch marked as taken?

**Comment by bence42** [comment]
Debugged a bit, and it turns out that the switch parsing logic is too complicated.
This handles all usecases, since all we care about is the exec count of the `case` line.

```python
        elif _is_a_switch(code):
            line_coverage.decision = DecisionCoverageSwitch(line_coverage.count)

```

**Comment by Spacetown** [comment]
https://github.com/gcovr/gcovr/blob/c83215a72f6ca434df6b483627efe76d6d53145e/gcovr/decision_analysis.py#L174-L184

In your `case 0` the `else` is used but an open brace doesn't have an execution count.
In your `case 1` also the `else` is used but this is an empty line with also no coverage.

It would be really difficult to handle all formatting styles

**Comment by Spacetown** [comment]
It seems, that this was fixed already in #738.
